src/service/resources-impl.js

```diff
@@ -1087,8 +1090,7 @@ export class Resources {
     const box = this.viewport_.getLayoutRect(element);
     const resource = Resource.forElement(element);
     if (box.width != 0 && box.height != 0) {
-      // TODO setRelayoutTop_ is being deprecated.
-      this.setRelayoutTop_(box.top);
+      this.dirtyElement(element);
```

Comment:
why is this change in this PR?

Reply:
These change should more-or-less equivalent. The call to `setRelayoutTop_` _could_ receive a different top value (the old one received the uncached box-top and the new one receives the cached box-top).

But if they were ever different, than it was a bug. The relayout top is always compared against cached values, never uncached.

Reply:
@jridgewell what happens if the `relayoutTop` is set by an element that's not in the scroll flow of the document? Could we end up laying out less elements or laying them out later than normal causing the dip?

I see `sticky ad` force calling `collapse` and ending up in here whenever an `amp-ad` inside `amp-sticky-ad` fails to loa, setting `layoutTop` to a fixed element's top which doesn't seem to be relevant to what `relayoutTop` is meant for.

(`amp-flying-carpet` is another one ends up here, surprisingly `attemptCollapse` which normal `amp-ad` calls does not seem to go through this code path).